src/Workspaces/Core/Portable/Workspace/Solution/Project.cs

```diff
@@ -287,9 +288,13 @@ public async ValueTask<IEnumerable<SourceGeneratedDocument>> GetSourceGeneratedD
             ImmutableInterlocked.GetOrAdd(ref _idToSourceGeneratedDocumentMap, state.Id, s_createSourceGeneratedDocumentFunction, (state, this)));
     }
 
-    internal async ValueTask<IEnumerable<Document>> GetAllRegularAndSourceGeneratedDocumentsAsync(CancellationToken cancellationToken = default)
+    internal async IAsyncEnumerable<Document> GetAllRegularAndSourceGeneratedDocumentsAsync([EnumeratorCancellation] CancellationToken cancellationToken)
     {
-        return Documents.Concat(await GetSourceGeneratedDocumentsAsync(cancellationToken).ConfigureAwait(false));
```

Comment:
old approach blocked the caller until SG docs were generated.  New approach allows the actual docs to be returned first.